packages/component/src/Activity/ScrollToEndButton.js

```diff
@@ -2,25 +2,49 @@ import { StateContext as ScrollToBottomStateContext } from 'react-scroll-to-bott
 
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
-import React, { useCallback } from 'react';
+import React, { useCallback, useRef } from 'react';
 
+import useActivities from '../hooks/useActivities';
 import useDirection from '../hooks/useDirection';
 import useFocusSendBox from '../hooks/useFocusSendBox';
 import useLocalize from '../hooks/useLocalize';
 import useScrollToEnd from '../hooks/useScrollToEnd';
 import useStyleSet from '../hooks/useStyleSet';
 
-const ScrollToEndButton = ({ className }) => {
-  const [direction] = useDirection();
+const ScrollToEndButton = ({ animating, className, sticky }) => {
   const [{ scrollToEndButton: scrollToEndButtonStyleSet }] = useStyleSet();
+  const [activities] = useActivities();
+  const [direction] = useDirection();
   const focusSendBox = useFocusSendBox();
+  const newMessageText = useLocalize('New messages');
   const scrollToEnd = useScrollToEnd();
 
   const handleClick = useCallback(() => {
     scrollToEnd();
     focusSendBox();
   }, [focusSendBox, scrollToEnd]);
 
+  // We ignore activity types other than "message"
+  const lastMessageActivity = [...activities].reverse().find(({ type }) => type === 'message');
+  const lastShownActivityId = (lastMessageActivity || {}).id;
+  const lastReadActivityIdRef = useRef(lastShownActivityId);
+
+  const { current: lastReadActivityId } = lastReadActivityIdRef;
+
+  if (sticky) {
+    // If it is sticky, mark the activity ID as read.
+    lastReadActivityIdRef.current = lastShownActivityId;
+  }
+
+  // Don't show the button if:
+  // - The scroll bar is animating
+  //   - If not set, will cause flashy button when: 1. Scroll to top, 2. Send something, 3. The button flash when it scrolling down
```

Comment:
```suggestion
  //   - Otherwise, this will cause a flashy button when: 1. Scroll to top, 2. Send something, 3. The button flashes when it is scrolling down
```